tiledb/test/src/attribute.rs

```diff
@@ -12,23 +14,41 @@ pub fn arbitrary_name() -> impl Strategy<Value = String> {
         )
 }
 
+fn arbitrary_fill_value<T>() -> impl Strategy<Value = T>
+where
+    T: Arbitrary,
+{
+    proptest::prelude::any::<T>()
+}
+
+pub fn arbitrary_for_datatype(
+    context: &Context,
+    datatype: Datatype,
+) -> impl Strategy<Value = TileDBResult<Attribute<'_>>> {
+    proptest::prelude::any::<bool>()
+        .prop_flat_map(move |nullable| {
+            (Just(nullable),
+            if nullable { proptest::prelude::any::<bool>().bind() } else { Just(false).bind() })
+                .prop_flat_map(move |(nullable, fill_nullable)| {
+                    fn_typed!(arbitrary_fill_value, datatype => {
+                        (arbitrary_name(), Just(nullable), crate::filter::arbitrary_list_for_datatype(context, datatype), arbitrary_fill_value, Just(fill_nullable))
```

Comment:
I know this is in a fn_typed! block so the formatter doesn't touch it. Could we pull out the arbitary_list_for_datatype into a temporary?

Reply:
Actually there's a different variant of `fn_typed` which does play well with the formatter, so I've changed this to use that instead.